### ISSUE TITLE
Chaturbate has empty categories

### ISSUE THREAD

**Issue by scorsese50** [dobbelina/repository.dobbelina#1206]
Chaturbate has empty categories. Version 1.1.91  Kodi 20.2   Nvidia Shield

**Comment by scorsese50** [comment]
Version 1.1.91

**Comment by scorsese50** [comment]
Yes, favorites still work.

**Comment by zzyuui** [comment]
Same here too.  No category results in Chaturbate. Running Kodi/Cumination on Amazon Fire TV stick and Android TV box (Mi Box). Neither one currently works.  Favorites are still working on both platforms.

**Comment by Gujal00** [comment]
User-agent issue
```
We noticed you're using a web browser that we don't support.
```

**Comment by TarkusLV** [comment]
CTB Recorder has the same issue, even though it's still able to record models already saved.  So I guess they changed something on their end.

**Comment by Gujal00** [comment]
The have switched from html to ajax api for the listings. Have to code for that

**Comment by foofoo11** [comment]
yup me too last few days now.. just blank when u hit a category.. version 1.1.91 on kodi 20.1
any issues? any fixes?

**Comment by foofoo11** [comment]
> The have switched from html to ajax api for the listings. Have to code for that

so what does that mean? what needs to be done to fix non working add-ons? thanks i'm a noob

**Comment by foofoo11** [comment]
> Same here too. No category results in Chaturbate. Running Kodi/Cumination on Amazon Fire TV stick and Android TV box (Mi Box). Neither one currently works. Favorites are still working on both platforms.

favorites didnt work for me.. i un-installed cumination and re-installed but still same.. i tried both dobblina and the crew.. both does the same.. just blank, no warnings or anything.. just empty

**Comment by scorsese50** [comment]
favorites work, nothing else

**Comment by dobbelina** [comment]
[Chaturbate Fix](https://github.com/dobbelina/repository.dobbelina/commit/d0c17cb2de612b8bded834cc7216fe943a191903)

**Comment by foofoo11** [comment]
> [Chaturbate Fix](https://github.com/dobbelina/repository.dobbelina/commit/d0c17cb2de612b8bded834cc7216fe943a191903)

just checked 5:11pm pacific time.. only tags show images.. all others still empty or blank.... do you have ETA on the fix? thanks

**Comment by zzyuui** [comment]
Help me out please.  How do you install the Chaturbate Fix? I installed Cumination from The Crew Repo. Currently running Cumination 1.1.91.
Thanks

**Comment by scorsese50** [comment]
They have to fix it I assume. On nest update.

**Comment by TarkusLV** [comment]
> > [Chaturbate Fix](https://github.com/dobbelina/repository.dobbelina/commit/d0c17cb2de612b8bded834cc7216fe943a191903)
> 
> just checked 5:11pm pacific time.. only tags show images.. all others still empty or blank.... do you have ETA on the fix? thanks

All categories I tried work now, but only when selected from the main Chaturbate menu.  They don't work when opening them from my custom menu.

**Comment by Gujal00** [comment]
> > > [Chaturbate Fix](https://github.com/dobbelina/repository.dobbelina/commit/d0c17cb2de612b8bded834cc7216fe943a191903)
> > 
> > 
> > just checked 5:11pm pacific time.. only tags show images.. all others still empty or blank.... do you have ETA on the fix? thanks
> 
> All categories I tried work now (with the fix installed), but only when selected from the main Chaturbate menu. They don't work when opening them from my custom menu.

You have to redo your custom menus as all menus are changed in chaturbate plugin

**Comment by pitsi** [comment]
The patch does not work on kodi 18, possibly due to python2 and its utf8 shortcomings. 
So here is a log. I removed the first ~600 lines from it because they are irrelevant and kept the remaining ~3700 ones that are related to the addon itself and its crash when selecting one category from cb.

[kodi.log](https://github.com/dobbelina/repository.dobbelina/files/13037752/kodi.log)

**Comment by pitsi** [comment]
The last cb patch did fix it and it now works again on kodi 18 and python2.

**Comment by psyco752** [comment]
I am on Kodi 19 and categories continue to appear and everything is still empty, is there an update on the way to solve it?

**Comment by marsrover1-ops** [comment]
Downloaded the 1.1.91 zip and incorporated the Chaturbate Fix listed above, however the search option for using custom keywords is not showing up.

**Comment by TarkusLV** [comment]
> Downloaded the 1.1.91 zip and incorporated the Chaturbate Fix listed above. All main menu options seem to work, however the search option for using custom keywords is not showing up. On firestick running Kodi 20

```
Only fixes listings and tags.
All other options commented out and would need fixing
```

**Comment by Gujal00** [comment]
> Downloaded the 1.1.91 zip and incorporated the Chaturbate Fix listed above. All main menu options seem to work, however the search option for using custom keywords is not showing up. On firestick running Kodi 20

Install [this patch](https://github.com/dobbelina/repository.dobbelina/commit/9982d911bd2a5cddc31fa6f2dc22fee0c73c5167) and test

**Comment by foofoo11** [comment]
> > Downloaded the 1.1.91 zip and incorporated the Chaturbate Fix listed above. All main menu options seem to work, however the search option for using custom keywords is not showing up. On firestick running Kodi 20
> 
> Install [this patch](https://github.com/dobbelina/repository.dobbelina/commit/9982d911bd2a5cddc31fa6f2dc22fee0c73c5167) and test

how do u install patch on firestick?? chaturbate still empty as of 10:30pm pacific

**Comment by dvgm506** [comment]
@foofoo11 Depends on what parts you want to fix, there's still work needing to be done for favorites (kinda) & followed. @dobbelina released 1.1.92 a couple hours after you posted with the listings/tags fixes. If it's not updating, Main menu -> addons -> my addons -> navigate left then check for updates.

Quick (or not so quick) fix for Favorites:

If you uncomment the commenting out for favorites, they still work. Guessing @dobbelina assumed they were broken and proactively disabled them. On my shieldtv, I downloaded the zip (or grab it from addons/packages, edited the chaturbate.py (line 50, got rid of the #) in the zip, copied to a usb stick then installed from zip as a temp measure until everything is fixed. Or on mac/windows/linux just edit the chaturbate.py directly (where I did my poking around).

BTW, dobbelina, thanks for all the hard work!
